### PR TITLE
Always go through the expn_that_defined query.

### PR DESCRIPTION
**Pull request by cjgillot**

### CODE REVIEW

**Comment by petrochenkov** [comment]
Minor regressions in incremental mode since we are tracking more, but this is necessary for correctness.
@bors r+

File: compiler/rustc_middle/src/query/mod.rs [hunk 1]
```diff
@@ -212,6 +212,7 @@ rustc_queries! {
 
     /// Internal helper query. Use `tcx.expansion_that_defined` instead
```
Comment (bjorn3) [inline compiler/rustc_middle/src/query/mod.rs:213]: Doc comment should be updated.

Comment (cjgillot) [inline compiler/rustc_middle/src/query/mod.rs:213]:
```suggestion
```

**Comment by cjgillot** [comment]
Just spoiled a CI cycle... not my brightest day :disappointed:

**Pull request merged**